### ISSUE TITLE
Key change separating back to back repeats.

### ISSUE THREAD

**Issue by stewkingjr** [musescore/MuseScore#20759]
### Issue type

Engraving bug (incorrect score rendering)

### Bug description

I have two sections that contain repeats (end repeat immediately followed by a start repeat of a new section), There us a key change in the new section. MuseScore inserts the new key in between the two repeat signs. It looks quite awkward and i have never seen that on a commercial published score.

### Steps to reproduce

1. Create a section with start repeat, measures, end repeat. 
2. Create additional measures which immediately begin with start repeat, some measures end repeat.
3. Put a key change in the beginning of the second section repeat group. 
4. Observe carefully the resulting engraving.

### MuseScore Version

It would be nice to be able to copy/paste this from MS. 4.2.0.233521124 ebad33c

### Regression

I don't know

### Operating system

win 11

### Additional context

![musescore](https://github.com/musescore/MuseScore/assets/22062172/1d9170d6-acfd-4697-a231-7e956e38d7e9)

**Comment by sammik** [comment]
It is "by design" as it follows Gould's rules.
I think, nowadays (almost) every software has it so. (Lilypond is light exception) .

But I think, it is really disgusting, and I hope, "classic" position could be added as an option.

Here is an example - Sammartini: Sinfonia in D (JC 14), Harvard University Press, Cambridge, Massachussets, 1968
![keysig-after-repeat](https://github.com/musescore/MuseScore/assets/1646034/91c9a30e-a559-475c-a3d6-691ad4aae87d)

(In this Symphony key signature changes back to major later in second repeat, so new style is not only ugly, but also incorrect there too, as if you go back to second repeat, key change is missing /as it is notated before repeat/).

**Comment by stewkingjr** [comment]
Thank you for the reply.

I don't know what other software does as i've decided musescore is where
i'm at. (I'm reminded of my mother who used to say, "And if your friends
say to jump off a bridge...?")

I've nearly 60 years on playing music, most of it in the classical arena
and have never, ever seen it treated that way. It's always been as you show
in the " keysig-after-repeat.png (view on web)
<https://github.com/musescore/MuseScore/assets/1646034/91c9a30e-a559-475c-a3d6-691ad4aae87d>
."
Here is an example, from a commercial publisher, of which i was editing:
Brahms hungarian dance 5.
[image: image.png]

Glen Gould (assuming that's to whom you refer) is a great pianist. Not sure
he's the authority on style despite what the musescore leaders might think.
Ok, maybe it's elaine gould in her book *behind bars*?

I tried everything i could think of, measures of zero beats (doesn't
actually happen), inserting a 1/128 measure and hiding it along with moving
the change in key signature and not counting the measure, etc, etc, and i
couldn't remove the ugliness. I don't have time to continue to play. But i
wonder what happens if a second ending starts with a key change? Same
thing? If not, why not?

When i'm questioned about it, i'll just say it's a crock of musescore.

Anyway, thanks again for the answer, and i hope you're able to lobby for
the traditional treatment.

Regards,
-stew

**Comment by oktophonie** [comment]
We just need an option for whether it goes before, between, or after. We don't need outraged declarations about it being "wrong" or "disgusting" or that "in my 600 years as a musician I've never seen anything as stupid as this", etc.

**Comment by stewkingjr** [comment]
I agree with you. My apologies. Here's what a commercial publisher does,
specifically N. Simrock Company. (see attached). Specific examples are on
their printed page numbers 9 and 22, but other examples are throughout..

If you'd like me to fill out a spec sheet, please forward one of those.

**Comment by sammik** [comment]
> I agree with you. My apologies. Here's what a commercial publisher does, specifically N. Simrock Company. (see attached). Specific examples are on their printed page numbers 9 and 22, but other examples are throughout.. If you'd like me to fill out a spec sheet, please forward one of those.

@stewkingjr please add attachments directly to github page, because if You attach it to message via mail, it is lost in translation.

**Comment by oktophonie** [comment]
Resolved via https://github.com/musescore/MuseScore/pull/26264 (implements toggle to configure whether changes should go between end-start repeats or not).
